pkg/controller/controllercmd/builder.go

```diff
@@ -251,12 +278,13 @@ func (b *ControllerBuilder) Run(ctx context.Context, config *unstructured.Unstru
 		if err != nil {
 			return err
 		}
-		serverConfig.Authorization.Authorizer = union.New(
-			// prefix the authorizer with the permissions for metrics scraping which are well known.
-			// openshift RBAC policy will always allow this user to read metrics.
-			hardcodedauthorizer.NewHardCodedMetricsAuthorizer(),
-			serverConfig.Authorization.Authorizer,
-		)
+
+		if b.authorizer != nil {
```

Comment:
so if it is nil, we get no authorizer?  that seems bad.

Reply:
this is never `nil` except unit test or when somebody generate operator without WithServer (which in that case it does not matter).